src/Module/User/Import.php

```diff
@@ -0,0 +1,415 @@
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->app            = $app;
+		$this->config         = $config;
+		$this->pconfig        = $pconfig;
+		$this->systemMessages = $systemMessages;
+		$this->database       = $database;
+		$this->permissionSet  = $permissionSet;
+	}
+
+	protected function post(array $request = [])
+	{
+		if ($this->config->get('config', 'register_policy') != \Friendica\Module\Register::OPEN && !$this->app->isSiteAdmin()) {
+			throw new HttpException\ForbiddenException($this->t('Permission denied.'));
+		}
+
+		$max_dailies = intval($this->config->get('system', 'max_daily_registrations'));
+		if ($max_dailies) {
+			$total = $this->database->count('user', ["`register_date` > UTC_TIMESTAMP - INTERVAL 1 DAY"]);
+			if ($total >= $max_dailies) {
+				throw new HttpException\ForbiddenException($this->t('Permission denied.'));
+			}
+		}
+
+		if (!empty($_FILES['accountfile'])) {
+			$this->importAccount($_FILES['accountfile']);
+		}
+	}
+
+	protected function content(array $request = []): string
+	{
+		if (($this->config->get('config', 'register_policy') != \Friendica\Module\Register::OPEN) && !$this->app->isSiteAdmin()) {
+			$this->systemMessages->addNotice($this->t('User imports on closed servers can only be done by an administrator.'));
+		}
+
+		$max_dailies = intval($this->config->get('system', 'max_daily_registrations'));
+		if ($max_dailies) {
+			$total = $this->database->count('user', ["`register_date` > UTC_TIMESTAMP - INTERVAL 1 DAY"]);
+			if ($total >= $max_dailies) {
+				$this->logger->notice('max daily registrations exceeded.');
+				$this->systemMessages->addNotice($this->t('This site has exceeded the number of allowed daily account registrations. Please try again tomorrow.'));
+			}
+		}
+
+		$tpl = Renderer::getMarkupTemplate('user/import.tpl');
+		return Renderer::replaceMacros($tpl, [
+			'$regbutt' => $this->t('Import'),
+			'$import'  => [
+				'title'    => $this->t('Move account'),
+				'intro'    => $this->t('You can import an account from another Friendica server.'),
+				'instruct' => $this->t('You need to export your account from the old server and upload it here. We will recreate your old account here with all your contacts. We will try also to inform your friends that you moved here.'),
+				'warn'     => $this->t("This feature is experimental. We can't import contacts from the OStatus network (GNU Social/Statusnet) or from Diaspora"),
+				'field'    => ['accountfile', $this->t('Account file'), '<input id="id_accountfile" name="accountfile" type="file">', $this->t('To export your account, go to "Settings->Export your personal data" and select "Export account"')],
+			],
+		]);
+	}
+
+	private function lastInsertId(): int
+	{
+		if (self::IMPORT_DEBUG) {
```

Comment:
This seems like a weird setup for debugging, do we really want it that way?

Reply:
Of course not, unit tests would be better, but heh.